etape1/src/main/java/org/iae/annecy/st1/etape1/view/PanierView.java

```diff
@@ -0,0 +1,48 @@
+package org.iae.annecy.st1.etape1.view;
+
+import org.iae.annecy.st1.etape1.controller.PanierController;
+import org.iae.annecy.st1.etape1.model.person.Person;
+import org.iae.annecy.st1.etape1.model.produit.Produit;
+
+public class PanierView{
+    
+    public PanierView(PanierController panierCont){
+	super();
+    }
+    
+    /**
+     * Affiche le panier avant validation
+     * 
+     * @param p
+     * 		le client à qui appartient le panier
+     * @return La liste des produits du panier 
+     */
+    public String afficherPanier(Person p){
+	String text = "Réf :\t\tNom :\t\tPrix :\t\tQté :\t\tPrix total par produit :\n";
+	double total = 0;
+	for (Produit produit : p.getPanier()) {
+	    text += produit.getReference() + "\t\t" + produit.getNom() + "\t\t" + produit.getPrix() + "€\t\t" + produit.getQuantite() + "\t\t" + 

```

Comment:
En Java il faut éviter le "+" avec les chaine de texte (dès que tu fait plus de concaténation sur une même chaine). Cela fonctionne bien mais est très consommateur en ressource (à cause l’immutabilité de String). Je te recommande d'utiliser un StringBufer à la place (https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuffer.html). Par "convention" on appel souvent la variable du StringBuffer "sb" (ce "sb" fait partie des "cas particulier" de non de variable "court acceptable").

- Déclare ton StringBuffer avant ta boucle "for".
- A la place de chaque "+" fait un "sb.append("xxxxxxxx");
- Remplace ton return par un `return sb.toString()`.